### ISSUE TITLE
CR IAC ignored in TelnetClient

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3305]
|<img alt="samp's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| samp reported|
|-|-|
|Trac ID|trac#3305|
|Type|defect|
|Created|2008-06-23 13:38:07Z|
|Branch|https://github.com/twisted/twisted/tree/cr-iac-3305|

If the server sends data like '\r\xff' (CR IAC), then Telnet ignores the IAC and puts it into the appData buffer. This is wrong, even if the server isn't obeying RFC 854.

<details><summary>Searchable metadata</summary>

```
trac-id__3305 3305
type__defect defect
reporter__samp samp
priority__normal normal
milestone__ 
branch__branches_cr_iac_3305 branches/cr-iac-3305
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1214228287000000 1214228287000000
changetime__1215615251000000 1215615251000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24224]) Branching to 'cr-iac-3305'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24225]) Handle CR IAC differently

refs [#3305](https://github.com/twisted/twisted/issues/3305)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [24226]) Perhaps provide a slightly more enlightening explanation for this choice

refs [#3305](https://github.com/twisted/twisted/issues/3305)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @exarkun|
|-|-|

Looks fine, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [24246]) Merge cr-iac-3305

Author: exarkun
Reviewer: therve
Fixes: [#3305](https://github.com/twisted/twisted/issues/3305)

Change `twisted.conch.telnet` to handle CR IAC as it would handle CR NUL IAC
instead of handling it as it would CR NUL IAC IAC, providing more useful
behavior for the case where a peer doesn't exactly follow the telnet RFC
with respect to the limitations on what can follow CR.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
